ospf6d/ospf6_top.c

```diff
@@ -1137,19 +1255,32 @@ static void ospf6_show(struct vty *vty, struct ospf6 *o, json_object *json,
 }
 
 /* show top level structures */
-DEFUN(show_ipv6_ospf6,
-      show_ipv6_ospf6_cmd,
-      "show ipv6 ospf6 [json]",
-      SHOW_STR
-      IP6_STR
-      OSPF6_STR
-      JSON_STR)
+DEFUN(show_ipv6_ospf6, show_ipv6_ospf6_cmd,
+      "show ipv6 ospf6 [vrf <NAME|all>] [json]",
+      SHOW_STR IP6_STR OSPF6_STR VRF_CMD_HELP_STR "All VRFs\n" JSON_STR)
 {
 	struct ospf6 *ospf6;
+	struct listnode *node;
+	const char *vrf_name = VRF_DEFAULT_NAME;
+	bool all_vrf = false;
+	int idx_vrf = 0;
+
 	bool uj = use_json(argc, argv);
 	json_object *json = NULL;
 
-	ospf6 = ospf6_lookup_by_vrf_name(VRF_DEFAULT_NAME);
+	OSPF6_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf);
+	if (all_vrf) {
+		for (ALL_LIST_ELEMENTS_RO(om6->ospf6, node, ospf6)) {
+			if (uj)
+				json = json_object_new_object();
+
+			ospf6_show(vty, ospf6, json, uj);
+		}
+		if (uj)
+			json_object_free(json);
+		return CMD_SUCCESS;
+	}
+	ospf6 = ospf6_lookup_by_vrf_name(vrf_name);
```

Comment:
Same comment as before. This call for specific VRFs could also have been accommodated inside the for loop for 'all_vrf'.

Reply:
Resolved.